### ISSUE TITLE
Add a way to ignore $ lines without uglifying the docs

### ISSUE THREAD

**Issue by hroncok** [pytest-dev/regendoc#9]
Currently, regendoc runs all commands from docs that start with `$`:

https://github.com/pytest-dev/regendoc/blob/27e3652723659d8b2d7a5311567480fccc49cbe2/regendoc/parse.py#L82-L84

In order to workaround this feature, the commands not desired regenerating are prepended with extra charters (such as `.` or `#`) in pytest documentation. This works (such lines are not regenerated), however such characters remains in the rendered HTML docs and are visible to the reader.

See https://github.com/pytest-dev/pytest/pull/5153

A better way (invisible to the user) to mark code snippet as skip-regenerating should probably be invented. A rst comment or directive might do.

**Comment by hroncok** [comment]
I've hit send too soon. Will edit.

**Comment by RonnyPfannschmidt** [comment]
@hroncok would you consider a hashtag as a valid way to handle this (aka use `$` to denote examples to run and `#` to denote examples to leave

an alternative would be a directive similar to wipe

currently regendoc is not integrated with restructuredtext/myst parsers as when i wrote it, i found no easy way to integrate it

**Comment by hroncok** [comment]
IIRC The main motivation for this report was that there are confusing characters in the documentation, not relevant to the reader. I am not sure mixing dollars and hashes for prompts will help the situation much.

**Comment by RonnyPfannschmidt** [comment]
@hroncok in part i'd be more than happy to port this to sphinx/myst in some way, but so far i haven`t approached a easy way to get the raw contents of the nodes i'd require to change and also changing them

**Comment by nicoddemus** [comment]
Perhaps changing from the current custom parser to proper directives would help, and open up more possibilites?

For example today `regendoc` interprets this:

```rst
To get pytest help, run:

    $ pytest --help
```

Perhaps we can change that to a doctest directive:

```rst
To get pytest help, run:

    .. regendoc-run: pytest --help
```

Which would replace the directive by what we have today. Both would output:

```rst
To get pytest help, run:

    $ pytest --help
    usage: pytest [options] [file_or_dir] [file_or_dir] [...]

    positional arguments:
      file_or_dir
    ...
```

**Comment by hroncok** [comment]
I like that approach.

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus this currently requires writing a own restructuredtext parser

this feature was initially planned for docutils sometime in 2008, as far as i understand it never landed to be able to round-trip/properly parse

basically - unless someone hands us a parser that can support the use-case i'm not going to do it

**Comment by nicoddemus** [comment]
> basically - unless someone hands us a parser that can support the use-case i'm not going to do it

Sure, I thought this was something ready in restructuredtext API itself. Do you mean every restructured text extension today parses the text manually?

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus no, the problem is that the restructured-text parser generates an ast, my understanding is, its not safely possible to write back a slightly changed file 

directives can choose whether they are raw or parsed

**Comment by nicoddemus** [comment]
Ahh right, it is a different use case. 👍 

Thanks for the clarification.
